Telemetry/RulesTest.cs

```diff
@@ -31,5 +60,315 @@ public void Should_Return_OK_Status()
             // Assert
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
         }
+
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void CreatesRuleWithInstantCalculation_IfValid()
+        {
+            // Arrange  
+            Assert.True(this.ValidChillerGroup()); // Make sure device groups have been created by seed data
+            var ruleRequest = this.GetSampleRuleWithCalculation("Instant", "0");
+
+            // Act
+            var request = new HttpRequest(Constants.TELEMETRY_ADDRESS + "/rules");
+            request.AddHeader("X-Foo", "Bar");
+            request.AddHeader("Content-Type", "application/json");
+            request.SetContent(JsonConvert.SerializeObject(ruleRequest));
+
+
+            var response = this.httpClient.PostAsync(request).Result;
+            var ruleResponse = JsonConvert.DeserializeObject<RuleApiModel>(response.Content);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(ruleRequest.Name, ruleResponse.Name);
+            Assert.Equal(ruleRequest.Description, ruleResponse.Description);
+            Assert.Equal(ruleRequest.GroupId, ruleResponse.GroupId);
+            Assert.Equal(ruleRequest.Severity, ruleResponse.Severity);
+            Assert.Equal(ruleRequest.Enabled, ruleResponse.Enabled);
+            Assert.Equal(ruleRequest.Calculation, ruleResponse.Calculation);
+            Assert.Equal(ruleRequest.Conditions[0].Field, ruleResponse.Conditions[0].Field);
+            Assert.Equal(ruleRequest.Conditions[0].Operator, ruleResponse.Conditions[0].Operator);
+            Assert.Equal(ruleRequest.Conditions[0].Value, ruleResponse.Conditions[0].Value);
+
+            this.rulesCreated.Add(ruleResponse.Id); // Track new rule for deletion
```

Comment:
this should happen before the assertions. One failing assertion will interrupt the method execution and this won't run